### PR TITLE
Add an API to track the status of the current page

### PR DESCRIPTION
**Pull request by jrochel**
depends on https://github.com/ocsigen/ocsigenserver/pull/133

### CODE REVIEW

File: src/lib/eliom_client.client.ml [hunk 1]
```diff
@@ -470,83 +470,104 @@ let url_fragment_prefix_with_sharp = "#!"
...
+    :: List.filter (fun (id', _) -> id <> id') !history_doms;
+  garbage_collect_cached_doms ()
+
+module Page_status = struct
+  include Page_status_t
+
+  let signal () = let p = get_this_page () in p.page_status
+
+  module Events = struct
+    let changes () = React.S.diff (fun target _ -> target) (signal ())
```
Comment (vouillon) [inline src/lib/eliom_client.client.ml]: `React.S.changes`

Comment (jrochel) [reply src/lib/eliom_client.client.ml]: Done.

File: src/lib/eliom_client.client.ml [hunk 2]
```diff
@@ -1253,19 +1287,23 @@ let () =
...
                 reload_function := Some rf;
                 let%lwt () = run_lwt_callbacks ev (flush_onchangepage ()) in
-                current_state_id := state_id;
-                rf () () >>
-                (scroll_to_fragment ~offset:state.position fragment;
-                 Lwt.return_unit)
+                let old_page = !active_page in
+                (*TODO: too early*)
+                set_active_page @@ mk_page ~state_id ~status:Generating ();
+                let%lwt () = rf () () in
```
Comment (vouillon) [inline src/lib/eliom_client.client.ml]: Shouldn't we evaluate `rf () ()` within `Lwt.with_value this_page !active_page`?

File: src/lib/eliom_client.client.ml [hunk 3]
```diff
@@ -1253,19 +1289,23 @@ let () =
                 let rf = List.assq state_id.state_index !reload_functions in
                 reload_function := Some rf;
                 let%lwt () = run_lwt_callbacks ev (flush_onchangepage ()) in
-                current_state_id := state_id;
-                rf () () >>
-                (scroll_to_fragment ~offset:state.position fragment;
-                 Lwt.return_unit)
+                let old_page = !active_page in
+                let new_page = mk_page ~state_id ~status:Generating () in
+                set_active_page new_page; (*TODO: too early but doesn't work otherwise *)
```
Comment (vouillon) [inline src/lib/eliom_client.client.ml]: Is this still true?

Comment (jrochel) [reply src/lib/eliom_client.client.ml]: Changed and tested.

**Pull request merged**